ios/Gekidou/Sources/Gekidou/Networking/ShareExtension.swift

```diff
@@ -36,18 +41,27 @@ struct UploadSessionData {
         let fileIds = dict["fileIds"] as! [String]
         let message = dict["message"] as! String
         let totalFiles = dict["totalFiles"] as! Int
+        let failNotificationID = dict["failNotificationID"] as! String
+        let failTimeout = dict["failTimeout"] as! Date
         
         return UploadSessionData(
             serverUrl: serverUrl,
             channelId: channelId,
             files: files,
             fileIds: fileIds,
             message: message,
-            totalFiles: totalFiles
+            totalFiles: totalFiles,
+            failNotificationID: failNotificationID,
+            failTimeout: failTimeout
         )
     }
 }
 
+// TODO: setup some configuration for allowing users to figure out the right ammount of time to wait.
```

Comment:
I don't think we should allow the user to set this, we should come up with a number that makes sense.. maybe an hour or so

Reply:
That's cool with me too. O just didn't feel entitled to make a decision on either option